From: Michal Hocko <mhocko@suse.cz>
Subject: Fix cmem_cgroup_cancel_carge units
Patch-mainline: No
References: bnc#697913

patches.suse/mm-thp-memcg-compound-reimplementation.patch introduced
a bug in mem_cgroup_cancel_charge where it changed units to be uncharged
from bytes to page count.

Let's remove the pointless page_size parameter in mem_cgroup_get_reclaim_stat_from_page
introduced by this patch as well.

Signed-off-by: Michal Hocko <mhocko@suse.cz>

Index: linux-2.6.32-thp/mm/memcontrol.c
===================================================================
--- linux-2.6.32-thp.orig/mm/memcontrol.c
+++ linux-2.6.32-thp/mm/memcontrol.c
@@ -862,10 +862,6 @@ mem_cgroup_get_reclaim_stat_from_page(st
 {
 	struct page_cgroup *pc;
 	struct mem_cgroup_per_zone *mz;
-	int page_size = PAGE_SIZE;
-
-	if (PageTransHuge(page))
-		page_size <<= compound_order(page);
 
 	if (mem_cgroup_disabled())
 		return NULL;
@@ -1497,9 +1493,9 @@ static void mem_cgroup_cancel_charge(str
 					int page_size)
 {
 	if (!mem_cgroup_is_root(mem)) {
-		res_counter_uncharge(&mem->res, page_size >> PAGE_SHIFT);
+		res_counter_uncharge(&mem->res, page_size);
 		if (do_swap_account)
-			res_counter_uncharge(&mem->memsw, page_size >> PAGE_SHIFT);
+			res_counter_uncharge(&mem->memsw, page_size);
 	}
 	css_put(&mem->css);
 }
